src/compression/mod.rs

```diff
@@ -0,0 +1,394 @@
+#[cfg(test)]
```

Comment:
This can be put behind an additional
```
#[cfg(any(
    feature = "zstd-compression",
    feature = "zlib-compression",
    feature = "snappy-compression"
))]
```

to remove the need for those within the test module.